src/l500/l500-color.cpp

```diff
@@ -221,7 +230,70 @@ namespace librealsense
                 return intrinsics;
             }
         }
-        throw std::runtime_error( to_string() << "intrinsics for resolution " << profile.width << "," << profile.height << " don't exist" );
+        throw std::runtime_error( to_string() << "intrinsics for resolution " << width << ","
+                                              << height << " don't exist" );
+    }
+
+    double l500_color_sensor::read_temperature() const
+    {
+        auto hwm = *_owner->_hw_monitor;
+
+        std::vector< byte > res;
+
+        try
+        {
+            res = hwm.send( command{ TEMPERATURES_GET } );
+        }
+        catch( std::exception const & e )
+        {
+            LOG_ERROR(
+                "Failed to get temperatures; hardware monitor in inaccessible: " << e.what() );
+            return 0.;
+        }
+
+        if( res.size() < sizeof( temperatures ) )  // New temperatures may get added by FW...
+        {
+            LOG_ERROR( "Failed to get temperatures; result size= "
+                       << res.size() << "; expecting at least " << sizeof( temperatures ) );
+            return 0.;
+        }
+        auto const & ts = *( reinterpret_cast< temperatures * >( res.data() ) );
+        LOG_DEBUG( "HUM temperture is currently " << ts.HUM_temperature << " degrees Celsius" );
+        return ts.HUM_temperature;
+    }
+
+     rs2_intrinsics normalize( const rs2_intrinsics & intr )
+    {
+        auto res = intr;
+        res.fx = 2 * intr.fx / intr.width;
+        res.fy = 2 * intr.fy / intr.height;
+        res.ppx = 2 * intr.ppx / intr.width - 1;
+        res.ppy = 2 * intr.ppy / intr.height - 1;
+
+        return res;
+    }
+
+    rs2_intrinsics
+    unnormalize( const rs2_intrinsics & intr, const uint32_t & width, const uint32_t & height )
```

Comment:
denormalize

Reply:
Shouldn't width and height be set to the intrinsics, too? If so, why not take them from there instead of passing them in?

Reply:
unnormalize gets normalized intrinsics and fit them to specific resolution